### ISSUE TITLE
How to have a uniform bottom navigator across the pages? I have the bottom navigator designed for 5 pages, but when i move to sub pages, how to retain the same bottom navigator? I want the same bottom navigator to stay for all the sub pages. Please help! Here's my code for Bottom Navigator.

### ISSUE THREAD

**Issue by Pushkar98** [flutter/flutter#16221]
import 'package:flutter/material.dart';
import 'package:ecommerce_pro/theme/style.dart';
import 'package:ecommerce_pro/screens/Home/index.dart' as HomeScreen;
import 'package:ecommerce_pro/screens/HomePro/index.dart' as HomePro;
import 'package:ecommerce_pro/screens/Categories/index.dart' as Categories;
import 'package:ecommerce_pro/screens/Bag/index.dart' as Bag;
import 'package:ecommerce_pro/screens/Notification/index.dart' as Notifications;

class HomeWithTab extends StatefulWidget {
  HomeWithTab({Key key}) : super(key: key);
  @override
  _HomeWithTabState createState() => new _HomeWithTabState();
}

class _HomeWithTabState extends State<HomeWithTab>
    with TickerProviderStateMixin {
  @override
  Widget build(BuildContext context) {
    TabController controller = new TabController(length: 6, vsync: this);
    Size screenSize = MediaQuery.of(context).size;
    return new Scaffold(
      backgroundColor: new Color.fromRGBO(255, 255, 255, 1.0),
      body: new TabBarView(
        children: <Widget>[
          new HomePro.HomePro(),
          new Categories.Categories(),
          new HomeScreen.HomeScreen(),
          new Notifications.Notifications(),
          new Bag.Bag(),
        ],
        controller: controller,
      ),
      bottomNavigationBar: new Container(
        padding: const EdgeInsets.only(
                  left: 0.0, right: 0.0, top: 0.0, bottom: .0),
        height: screenSize.height / 9.0,
        width:screenSize.width/1.0,
        decoration: new BoxDecoration(
            ),
        child: new TabBar(
          tabs: <Tab>[
            new Tab(
              child: new Column(
                children: <Widget>[
                  new Icon(
                    Icons.home,
                    size: 25.0,
                  ),
                  new Text("Home",
                   style: new TextStyle(
                  ))
                ],
              ),
            ),
            new Tab(
              child: new Column(
                children: <Widget>[
                  new Icon(
                    Icons.category,
                    size: 25.0,
                  ),
                  new Text("Categories",
                   style: new TextStyle(
                  ))
                ],
              ),
            ),
            new Tab(
              child: new Column(
                children: <Widget>[
                  new Icon(
                    Icons.account_box,
                    size: 25.0,
                  ),
                  new Text("Profile",
                   style: new TextStyle(
                  ))
                ],
              ),
            ),
            new Tab(
              child: new Column(
                children: <Widget>[
                  new Icon(
                    Icons.notifications_none,
                    size: 25.0,
                  ),
                  new Text("Notification",
                  style: new TextStyle(
                  ))
                ],
              ),
            ),
            new Tab(
              child: new Column(
                children: <Widget>[
                  new Icon(
                    Icons.card_travel,
                    size: 25.0,
                  ),
                  new Text("Bag",
                   style: new TextStyle(
                  ))
                ],
              ),
            ),
          ],
          indicatorColor:indicatorColor,   
          labelColor: indicatorColor,  
          unselectedLabelColor: Colors.black45,
          unselectedLabelStyle: new TextStyle(
            fontWeight: FontWeight.normal, 
            color: const Color.fromRGBO(119, 119, 119, 1.0),
            fontSize: 9.0),
          controller: controller,
          labelStyle: new TextStyle(
            fontSize: 9.0,
            color: const Color.fromRGBO(119, 119, 119, 1.0),
           fontWeight: FontWeight.bold,
          ),
        ),
      ),
    );
  }
    }

**Comment by anotherHarry** [comment]
https://github.com/flutter/flutter/issues/16181

tl;dr: Use CupertinoTabBar with CupertinoTabScaffold

The problem is not in Flutter but in UX just like Rémi Rousselet has mentioned.

It turned out Material Design doesn't recommend sub-pages in the hierarchy to access the Bottom navigation bar.

However, iOS Human Interface Guide recommend this. So, to use this feature, I had to adapt Cupertino widgets instead of Material ones. Specifically, in main, return a WidgetsApp/MaterialApp which contains a CupertinoTabScaffold. Implement the tab bar with a  CupertinoTabBar and each screen is a CupertinoTabView.

**Comment by mohith7548** [comment]
I'm sorry to say that you are implementing bottom navigation bar incorrectly. You are using tabs in bottom navigation bar which is wrong. I understood your problem and I have implemented it easily and correct way. see this code below. 

```import 'package:flutter/material.dart';
import 'package:firebase_auth/firebase_auth.dart';
import 'package:my_nit2018/navigarion_drawer.dart';
import 'package:my_nit2018/pages/app/blog/blog_page.dart';
import 'package:my_nit2018/pages/app/home/home_page.dart';
import 'package:my_nit2018/pages/app/library/library_page.dart';
import 'package:my_nit2018/pages/app/notifications/notifications_page.dart';

class MainApp extends StatefulWidget {
  FirebaseUser user;

  MainApp(this.user);

  @override
  _MainAppState createState() => new _MainAppState();
}

class _MainAppState extends State<MainApp> {
  int i = 0;
  var pages = [
    new HomePage(),
    new BlogPage(),
    new LibraryPage(),
    new NotificationsPage()
  ];

  @override
  Widget build(BuildContext context) {
    return new Scaffold(
      body: pages[i],
      drawer: new AppNavigationDrawer(),
      bottomNavigationBar: new BottomNavigationBar(
        items: [
          new BottomNavigationBarItem(
            icon: new Icon(Icons.home),
            title: new Text('Home'),
          ),
          new BottomNavigationBarItem(
            icon: new Icon(Icons.photo_library),
            title: new Text('Blog'),
          ),
          new BottomNavigationBarItem(
            icon: new Icon(Icons.book),
            title: new Text('Library'),
          ),
          new BottomNavigationBarItem(
            icon: new Icon(Icons.notifications),
            title: new Text('Notifications'),
          ),
        ],
        currentIndex: i,
        type: BottomNavigationBarType.fixed,
        onTap: (index) {
          setState(() {
            i = index;
          });
        },
      ),
    );
  }
}
```

**Comment by jcodinginflow** [comment]
Hi I am not sure this question relates here but I seriously need to know how to have a backdrop and also avigation bar at same time on the same page. I tried to write the code but I am stuck at having two "body" in same Scaffold

**Comment by FullstackJack** [comment]
@kcatstack @mohith7548 No offense and maybe things have changed since the time this question was answered but Material Design does very specifically call for sub-navigation within Bottom Navigation now. Drawer menus are less useful for top-level nav due to low visibility and just look at Instagram on Android. They use bottom tabs with sub-navigation that persists tab state when you navigate to photos and profiles within Explore. It's a great user experience.

See Behavior guidance from Google itself.
https://material.io/design/components/bottom-navigation.html#behavior

**Comment by FullstackJack** [comment]
Please implement the sub-navigation Bottom Navigation behavior as outlined in the spec. You can feature flag it if that makes it easier. persistNavigationBar?

https://material.io/design/components/bottom-navigation.html#behavior

**Comment by Shiiboun** [comment]
I hesitated to reopen this but are there any news on this? @FullstackJack have you been able to fix this?

**Comment by pinpong** [comment]
Anyone here with a good solution?
